pkg/datapath/linux/route/route_linux.go

```diff
@@ -514,6 +514,18 @@ func lookupDefaultRoute(family int) (netlink.Route, error) {
 	return routes[0], nil
 }
 
+func DeleteRouteTable(table, family int) error {
+	routes, err := netlink.RouteListFiltered(family, &netlink.Route{Table: table}, netlink.RT_FILTER_TABLE)
+	if err != nil {
+		return fmt.Errorf("Unable to list table %d routes: %s", table, err)
+	}
+
+	for _, route := range routes {
+		netlink.RouteDel(&route)
```

Comment:
Is this really error-free? Seems like we shouldn't be ignoring the error here, even if it's a rare case like some administrator messing with route tables while we run this code from Cilium.

Reply:
^ Probably good enough just to log it here at Warning level and proceed, since overall we just want to make the best effort to delete all of the relevant routes/table and move on.

Reply:
Maybe one more level of smartness, if we can do an `errors.Is(err, unix.ENOENT)`  then happy to ignore the error. If it's *not* ENOENT then I think we should be more concerned since the route wasn't deleted.

Reply:
Pushes a warning now if route was not deleted. I'm not sure when a ENENT would happen her we are walking the we just fetched.